### PR TITLE
Update docs to dioxus next

### LINKED ISSUE #139: The deployed website is crashing from today
The docsite was working until today.

In the browser console (macOS; Brave; guard disabled), I see:

```
docsite.js:725 ERROR /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/dioxus-web-0.4.0/src/lib.rs:219 Rehydration failed NodeNotFound. Rebuild DOM into element from scratch
inline0.js:255 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'listening')
    at run (inline0.js:255:330)
    at dioxus_interpreter_js::sledgehammer_bindings::Channel::flush::h3dd42f069636a006 (docsite_bg.wasm:0xaa7cd)
    at dioxus_web::dom::WebsysDom::apply_edits::h8dea4df7adec0978 (docsite_bg.wasm:0x7f70a)
    at dioxus_web::run_with_props::{{closure}}::hd8fbd61113cac6f4 (docsite_bg.wasm:0x7ed66)
    at wasm_bindgen_futures::queue::Queue::new::{{closure}}::h4af736430e73b26a (docsite_bg.wasm:0xe5a45)
    at <dyn core::ops::function::FnMut<(A,)>+Output = R as wasm_bindgen::closure::WasmClosure>::describe::invoke::h9e889b7261c861ea (docsite_bg.wasm:0x175211)
    at __wbg_adapter_63 (docsite.js:332:10)
    at real (docsite.js:236:20)

```

image

### LINKED ISSUE #200: Directly Navigating to /learn Redirects to /learn/0.4 But Then Crashes
# Issue
When navigating to the ``dioxuslabs.com/learn`` directly, the URL is redirected correctly to ``dioxuslabs.com/learn/0.4`` however the app panics.

**Images**

This is on Linux Mint, Firefox.
image

This is on Windows 10, Chrome.
image

### PR DESCRIPTION
**Pull request by ealmloff**
This PR updates the docs to match the changes in https://github.com/DioxusLabs/dioxus/pull/1791 along with other 0.5 changes

Closes #200
Closes #139

TODO:
- [x] Pull out and update examples in the error handling page
- [x] Update the tailwind/assets docs
~- [ ] Update and check the mobile docs~ Every version of wry after 0.34 is broken on mobile
- [x] Create a migration guide
- [x] Replace use_future with use_resource
- [x] Pull out examples in the router reference
- [x] Pull out examples in the ssr reference
- [x] Pull out examples in the use effect reference
- [x] Final run through of every doc page

### CODE REVIEW

File: docs-src/0.4/en/reference/component_props.md [hunk 1]
```diff
@@ -131,8 +131,8 @@ struct TitleCardProps {
     title: String,
 }
 
-fn TitleCard(cx: Scope<TitleCardProps>) -> Element {
-    cx.render(rsx!{
+fn TitleCard(<TitleCardProps>) -> Element {
+    rsx!{
         h1 { "{cx.props.title}" }
     })
 }
```
Comment (marc2332) [inline docs-src/0.4/en/reference/component_props.md:138]:
```suggestion
fn TitleCard(props: TitleCardProps) -> Element {
    rsx!{
        h1 { "{props.title}" }
    })
}
```

File: docs-src/0.4/en/reference/component_props.md
Comment (marc2332) [inline docs-src/0.4/en/reference/component_props.md:1]: github doesn't let me review an unchanged line, but anyway, this goes in L130
```rs
#[derive(Props, Clone, PartialEq)]
struct TitleCardProps {
    title: String,
}
```

**Review by marc2332 — CHANGES_REQUESTED** [review]

File: docs-src/0.4/en/reference/component_props.md [hunk 2]
```diff
@@ -121,19 +121,19 @@ Then, you can use it so:
...
     title: String,
 }
 
-fn TitleCard(cx: Scope<TitleCardProps>) -> Element {
-    cx.render(rsx!{
-        h1 { "{cx.props.title}" }
+fn TitleCard(props: TitleCardProps) -> Element {
+    rsx!{
+        h1 { "{props.title}" }
     })
```
Comment (marc2332) [inline docs-src/0.4/en/reference/component_props.md:137]: just noticed this sorry
```suggestion
    }
```

**Review by marc2332 — CHANGES_REQUESTED** [review]

**Comment by ealmloff** [comment]
Bumping the dioxus version solves several hydration issues:
Closes #200
Closes #139

**Comment by ealmloff** [comment]
Dioxus mobile and wry mobile is currently broken with any wry version of `>0.34` (https://github.com/tauri-apps/cargo-mobile2/issues/267)

**Comment by ealmloff** [comment]
The content is done, but I'm marking this as a draft until I get the static generation working again

**Comment by ealmloff** [comment]
Prebuilds are working and do not override the previous doc versions

File: docs-src/0.5/en/getting_started/mobile.md [hunk 3]
```diff
@@ -94,64 +94,100 @@ cargo mobile init
...
-            <div id="main"></div>
-            <!-- MODULE LOADER -->
-          </body>
-        </html>
-       "#.into()),
+#[cfg(target_os = "android")]
+fn init_logging() {
+    android_logger::init_once(
+        android_logger::Config::default()
+            .with_min_level(log::Level::Trace)
```
Comment (ritzk) [inline docs-src/0.5/en/getting_started/mobile.md:120]:
```suggestion
            .with_max_level(log::LevelFilter::Trace)
```

File: docs-src/0.5/en/getting_started/mobile.md [hunk 4]
```diff
@@ -94,64 +94,100 @@ cargo mobile init
 
 When you run `cargo mobile init`, you will be asked a series of questions about your project. One of those questions is what template you should use. Dioxus currently doesn't have a template in Tauri mobile, instead you can use the `wry` template.
 
-> You may also be asked to input your team ID for IOS. You can find your team id [here](https://developer.apple.com/help/account/manage-your-team/locate-your-team-id/) or create a team id by creating a developer account [here](https://developer.apple.com/help/account/get-started/about-your-developer-account)
+> You may also be asked to input your team ID for IOS. You can find your team id [here](https://developer.apple.com/help/account/manage-your-team/locate-your-team-id/) or create a team id by creating a develwr; oper account [here](https://developer.apple.com/help/account/get-started/about-your-developer-account)
```
Comment (ritzk) [inline docs-src/0.5/en/getting_started/mobile.md:97]:
```suggestion
> You may also be asked to input your team ID for IOS. You can find your team id [here](https://developer.apple.com/help/account/manage-your-team/locate-your-team-id/) or create a team id by creating a developer account [here](https://developer.apple.com/help/account/get-started/about-your-developer-account)
```

File: docs-src/0.5/en/getting_started/mobile.md (hunk 3 above)
Comment (ealmloff) [reply docs-src/0.5/en/getting_started/mobile.md]: Good catch, thanks!

**Pull request merged**